color/src/css.rs

```diff
@@ -127,6 +127,22 @@ impl CssColor {
         }
     }
 
+    /// Clip the color's components to fit within the natural gamut of the color space, and clamp
+    /// the color's alpha to be in the range `[0, 1]`.
+    ///
+    /// See [`ColorSpace::clip`] for more details.
+    #[must_use]
+    pub fn clip(self) -> Self {
+        let (opaque, alpha) = split_alpha(self.components);
+        let components = self.cs.clip(opaque);
+        let alpha = alpha.clamp(0., 1.);
```

Comment:
Would it be meaningful to have a separate `clip_alpha` method? Semantically, a less than zero alpha is definitely equivalent to zero, and likely similarly for greater than 1?

Reply:
Maybe; would `clip` leave alpha unchanged in that case?

Reply:
I don't think so? The other thing to think about is clipping the alpha for premultiplied

Reply:
I think that makes a good case to add it as a convenience method. Clipping unconditionally to premultiply would be a performance hit (plus premul math still works out when out-of-bounds, if `alpha` is interpreted as an arbitrary weighting).